fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -874,6 +875,17 @@ public Collection<Partition> getAllPartitions() {
         return partitions;
     }
 
+    public Collection<Partition> getRecentPartitions(int recentPartitionNum) {
+        List<Partition> partitions = Lists.newArrayList(idToPartition.values());
+        Collections.sort(partitions, new Comparator<Partition>() {
+            @Override
+            public int compare(Partition h1, Partition h2) {
+                return (int) (h2.getVisibleVersion() - h1.getVisibleVersion());
+            }
+        });
+        return partitions.subList(0, recentPartitionNum - 1);
```

Comment:
recentPartitionNum - 1 ==> recentPartitionNum?

end index is not included in subLIst.